### PR TITLE
[WIP] fix/bokeh_whitelist

### PR DESCRIPTION
**Pull request by quasiben**
PR fixes a small annoyance with whitelisting for distributed.  Previously, if one type

```
dask-scheduler --bokeh-port=9000 --bokeh-whitelist="*"
```

This would only proxy for host connections on `"*:9000`.  If a user is proxied (as often is the case inside docker containers) then it is likely they will be hitting some other host *usually* on port 80.  With this PR, a user can now type the following

```
dask-scheduler --bokeh-port=9000 --bokeh-whitelist="*" --prefix=/some_prefix
```

And expect all `host:port` combos to be available hosts and an optional prefix

### CODE REVIEW

**Comment by quasiben** [comment]
Happy to fix tests but perhaps my solution is undesirable since now `--bokeh-whitelist` will need to include the `port` if defining a specific `host`

`--bokeh-whitelist='127.9.8.10:9000'`

**Comment by mrocklin** [comment]
@quasiben what is the right thing to do here?

**Comment by quasiben** [comment]
@mrocklin, I'd advocate for my PR -- we want to allow for users to control both IP and port

**Comment by mrocklin** [comment]
Question for you.  Is there an input we can add to just always whitelist everything all the time and do away with all this?

**Comment by quasiben** [comment]
Would you also want x-headers turned on as well ?

**Comment by mrocklin** [comment]
I don't know what that means.  Let me rephrase.  Is there an appropriate nuclear option to get rid of this problem for good?  This kind of solution might not be appropriate given Bokeh's security guarantee's but may be appropriate given dask.distributed's weaker guarantees.

**Comment by mrocklin** [comment]
OK, sounds like I should rebase and merge?  I'm mostly just trusting you to know if this should go in or not.

**Comment by quasiben** [comment]
I don't think there is an appropriate nuclear option.  Behind proxies you will often still need to define a prefix.

**Comment by quasiben** [comment]
I think this should go and feel free to ping me if this becomes an issue

**Comment by mrocklin** [comment]
Rebased.  @quasiben can you take a look and verify that this is what you intended?

**Comment by quasiben** [comment]
yup, looks good
